### PR TITLE
golangci-lint: enable containedctx

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-3203

This PR enabled the `containedctx` rule and cleans up all outstanding cases.
> containedctx is a linter that detects struct contained context.Context field. This is discouraged technique in favour of passing context as first argument of method or function. For rationale please read [Contexts and structs](https://go.dev/blog/context-and-structs) the Go blog post.

It also cleans up a few other simple linter violations.

### CODE REVIEW

**Review by patrickhuie19 — APPROVED** [review]

**Comment by jmank88** [comment]
Rebased and needed to run `generate`, which crashed, so I updated mockery as well.

The tidy step flagged the generate problem but was not blocking merge.

**Comment by patrickhuie19** [comment]
> The tidy step flagged the generate problem but was not blocking merge.

Is there a version difference between CI and your local setup? looks like the step is failing, but on whitespace differences

**Comment by jmank88** [comment]
> > The tidy step flagged the generate problem but was not blocking merge.
> 
> Is there a version mismatch between CI and your local setup? looks like the step is failing, but on whitespace differences

Not a mismatch, but it was still failing to run fully. And I see the root of the issue: we generate code via a go template which fails the linter. Should be easy to fix the template.

File: pkg/codec/encodings/binary/gen/ints_test.go.tmpl [hunk 1]
```diff
@@ -187,7 +187,7 @@ func TestUint{{.}}(t *testing.T) {
 {{ end }}
 
 func TestBuilderInt(t *testing.T) {
-    {{ range . }}
+    {{ range . -}}
```
Comment (jmank88) [inline pkg/codec/encodings/binary/gen/ints_test.go.tmpl:190]: This prevents incidental whitespace.

**Review by patrickhuie19 — APPROVED** [review]

**Review by krehermann — DISMISSED** [review]

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**